File: WorldConquest/src/models/Province.ts
// Province Model

/**
 * Represents a province in the game
 * Each province is identified by its RGB color value from the map bitmap
 */
export class Province {
    // Province ID in the format R_G_B from the bitmap
    id: string;
    
    // Province name
    name: string;
    
    // Owner country ID, null if unowned
    ownerId: string | null;
    
    // Province type (land, water, wasteland)
    type: ProvinceType;
    
    // Province development level (countryside, village, town, city)
    level: ProvinceLevel;
    
    // Religion of the province
    religion: string;
    
    // Unrest level (0-100)
    unrest: number;
    
    // Buildings in the province
    buildings: Building[];
    
    // Armies stationed in the province
    armies: Army[];
    
    constructor(id: string, name: string, type: ProvinceType = ProvinceType.Land) {
        this.id = id;
        this.name = name;
        this.ownerId = null;
        this.type = type;
        this.level = ProvinceLevel.Countryside;
        this.religion = "";
        this.unrest = 0;
        this.buildings = [];
        this.armies = [];
    }
    
    // Get garrison strength based on province level
    get garrisonStrength(): number {
        return this.level * 100;
    }
    
    // Get income generated per turn
    get income(): number {
        return this.level * 50;
    }
}

// Province type enum
export enum ProvinceType {
    Land,
    Water,
    Wasteland
}

// Province level enum
export enum ProvinceLevel {
    Countryside = 1,
    Village = 2,
    Town = 3,
    City = 4,
    Metropolis = 5
}

// Import related models
import { Building } from './Building';
import { Army } from './Army';

File: WorldConquest/Models/Province.cs
using System.Collections.Generic;

namespace WorldConquest.Models
{
    public enum ProvinceLevel
    {
        Countryside,
        Village,
        Town,
        City
    }

    public enum ProvinceType
    {
        Land,
        Water,
        Wasteland
    }

    public class Province
    {
        // Province Id in the form "R_G_B"
        public string Id { get; set; }

        // Country Id or null if unowned
        public string? OwnerId { get; set; }

        // List of Army Ids or Unit Ids present in the province
        public List<string> TroopIds { get; set; } = new List<string>();

        // List of Building Ids present in the province
        public List<string> BuildingIds { get; set; } = new List<string>();

        // Province level (countryside, village, town, city, etc)
        public ProvinceLevel Level { get; set; }

        // Religion of the province
        public string Religion { get; set; }

        // Province type: Land, Water, or Wasteland
        public ProvinceType Type { get; set; }

        // Unrest value (0 = no unrest, higher = more unrest)
        public int Unrest { get; set; }

        // Whether the province can be upgraded (inferred from level/money in logic)
        public bool Upgradable { get; set; }
    }
}